### DISCUSSION TITLE
v8 native functions

### DISCUSSION THREAD

**Discussion by Eomm** [biomejs/biome#3132] [Q&A] [answered]
Give this script that uses the [`--allow-natives-syntax`](https://v8.dev/docs/builtin-functions) v8 flag:

```bash
node --allow-natives-syntax file.js
```

```js
'use strict'
// file.js

const a = {}
const b = {}

console.log(%HaveSameMap(a, b)) // <---- native syntax here
```

I can't setup Biome to recognize the `%HaveSameMap` syntax:

> Runtime functions are written in C++ and called from JavaScript through a %-prefix

<img width="561" alt="image" src="https://github.com/biomejs/biome/assets/11404065/daa9b28a-17e3-487f-9e36-67f8c5643311">

Plus:

```
  ✖ Expected an expression for the left hand side of the `%` operator.
  
    4 │ const b = {}
    5 │ 
  > 6 │ console.log(% HaveSameMap(a, b))
      │             ^
    7 │ 
  
  ℹ This operator requires a left hand side value
```

Do you know how to configure it?

**Comment by arendjr** [comment, marked as answer]
I’m afraid this is not currently supported by Biome, since the syntax is not valid JS/TS.

We probably *could* support it, similar to how we support JSX too, but it’s not currently implemented.

**Comment by Conaclos** [comment]
This is an ivalid ECMAScript syntax. I am not sure if we want to support this. What about the behavior of other tools?
